Sources/Sentry/SentryTracer.m

```diff
@@ -101,9 +104,117 @@ @implementation SentryTracer {
 #endif // SENTRY_HAS_UIKIT
 }
 
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+static NSObject *appLaunchTraceLock;
+static SentryId *_Nullable appLaunchTraceId;
+BOOL isTracingAppLaunch;
+static uint64_t appLaunchSystemTime;
+SentryTracesSamplerDecision *appLaunchTraceSamplerDecision;
+
+/**
+ * Decision tree:
+ * ```
+ * ─ is there a configuration file?
+ *   ├── yes: SentryTracesSampler sampling decision?
+ *   │   ├── yes: SentryProfilesSampler sampling decision?
+ *   │   │   ├── yes: ✅ trace app launch
+ *   │   │   └── no: ❌ do not trace app launch
+ *   │   └── no: ❌ do not trace app launch
+ *   └── no: ❌ do not trace app launch
+ * ```
+ */
++ (BOOL)shouldTraceAppLaunch
+{
+    [SentryFileManager load];
+
+    NSString *configFilePath = [SentryFileManager sentryLaunchConfigPath];
+
+    if (![[NSFileManager defaultManager] fileExistsAtPath:configFilePath]) {
+        SENTRY_LOG_DEBUG(@"No launch profile config file.");
+        return NO;
+    }
+
+    // get app launch configuration options
+    NSError *error;
+    NSData *configData = [NSData dataWithContentsOfFile:configFilePath options:0 error:&error];
+    if (error != nil) {
+        // we can't read the config file? bail out
+        SENTRY_LOG_DEBUG(@"Couldn't read launch profile config file: %@.", error);
+        return NO;
+    }
+
+    if (configData == nil) {
+        SENTRY_LOG_DEBUG(@"The profile config file was empty.");
+        return NO;
+    }
+
+    error = nil;
+    // a map of strings to numeric values for sample rates or boolean values for
+    // enabling/disabling launch tracing
+    NSDictionary<NSString *, NSNumber *> *launchConfig =
+        [NSJSONSerialization JSONObjectWithData:configData options:0 error:&error];
+    if (error != nil) {
+        SENTRY_LOG_DEBUG(@"Couldn't deserialize launch profile config file: %@", error);
+        return NO;
+    }
+
+    double tracesSampleRate = [launchConfig[@"tracesSampleRate"] doubleValue];
+    double profilesSampleRate = [launchConfig[@"profilesSampleRate"] doubleValue];
+    if (tracesSampleRate == 0 || profilesSampleRate == 0) {
+        SENTRY_LOG_DEBUG(@"Sampling out this launch trace due to sample rate of 0.");
+        return NO;
+    }
+
+    [SentryTracesSampler load];
+    [SentryProfilesSampler load];
+    [SentryRandom load];
+
+    SentryRandom *random = [[SentryRandom alloc] init];
+    appLaunchTraceSamplerDecision = [SentryTracesSampler calcSample:tracesSampleRate random:random];
+    if (appLaunchTraceSamplerDecision.decision != kSentrySampleDecisionYes) {
```

Comment:
`h`: The `SentrySDK.configureLaunchProfiling` could already store the sampling decision instead of the sample rate, so we minimize the work here. So instead of loading the sample rates here and then doing the calculations, the SentrySDK.configureLaunchProfiling` does the calculations and stores a single value if the app launch profiling should run or not. We could even take it a step further and only add a marker file when the app launch profiling should run. If the file doesn't exist, we skip profiling. Then we don't need to serialize the contents here.
Or am I missing something?

Reply:
I went a little bit further and am just writing an empty marker file now, and simply check for its existence.